src/ML-POD/eapod.cpp

```diff
@@ -0,0 +1,4734 @@
+      if (keywd == "fourbody_number_radial_basis_functions")
+        nrbf4 = utils::inumeric(FLERR,words[1],false,lmp);
+      if (keywd == "fourbody_angular_degree")
+        P4 = utils::inumeric(FLERR,words[1],false,lmp);
+      if (keywd == "fivebody_number_radial_basis_functions")
+        nrbf33 = utils::inumeric(FLERR,words[1],false,lmp);
+      if (keywd == "fivebody_angular_degree")
+        P33 = utils::inumeric(FLERR,words[1],false,lmp);
+      if (keywd == "sixbody_number_radial_basis_functions")
+        nrbf34 = utils::inumeric(FLERR,words[1],false,lmp);
+      if (keywd == "sixbody_angular_degree")
+        P34 = utils::inumeric(FLERR,words[1],false,lmp);
+      if (keywd == "sevenbody_number_radial_basis_functions")
+        nrbf44 = utils::inumeric(FLERR,words[1],false,lmp);
+      if (keywd == "sevenbody_angular_degree")
+        P44 = utils::inumeric(FLERR,words[1],false,lmp);
+    }
+  }
+  // if (nrbf2 < nrbf3) error->all(FLERR,"number of three-body radial basis functions must be equal or less than number of two-body radial basis functions");
+  if (nrbf3 < nrbf4) error->all(FLERR,"number of four-body radial basis functions must be equal or less than number of three-body radial basis functions");
+  if (nrbf4 < nrbf33) error->all(FLERR,"number of five-body radial basis functions must be equal or less than number of four-body radial basis functions");
+  if (nrbf4 < nrbf34) error->all(FLERR,"number of six-body radial basis functions must be equal or less than number of four-body radial basis functions");
+  if (nrbf4 < nrbf44) error->all(FLERR,"number of seven-body radial basis functions must be equal or less than number of four-body radial basis functions");
+  nrbfmax = (nrbf2 < nrbf3) ? nrbf3 : nrbf2;
+  nrbfmax = (nrbfmax < nrbf4) ? nrbf4 : nrbfmax;
+  nrbfmax = (nrbfmax < nrbf33) ? nrbf33 : nrbfmax;
+  nrbfmax = (nrbfmax < nrbf34) ? nrbf34 : nrbfmax;
+  nrbfmax = (nrbfmax < nrbf44) ? nrbf44 : nrbfmax;
+
+  if (P3 < P4) error->all(FLERR,"four-body angular degree must be equal or less than three-body angular degree");
+  if (P4 < P33) error->all(FLERR,"five-body angular degree must be equal or less than four-body angular degree");
+  if (P4 < P34) error->all(FLERR,"six-body angular degree must be equal or less than four-body angular degree");
+  if (P4 < P44) error->all(FLERR,"seven-body angular degree must be equal or less than four-body angular degree");
+
+  if (P3 > 12) error->all(FLERR,"three-body angular degree must be equal or less than 12");
+  if (P4 > 6) error->all(FLERR,"four-body angular degree must be equal or less than 6");
+
+  int Ne = nelements;
+  memory->create(elemindex, Ne*Ne, "elemindex");
+  int k = 0;
+  for (int i1 = 0; i1<Ne; i1++)
+    for (int i2 = i1; i2<Ne; i2++) {
+      elemindex[i2 + Ne*i1] = k;
+      elemindex[i1 + Ne*i2] = k;
+      k += 1;
+    }
+
+  init2body();
+  init3body(P3);
+  init4body(P4);
+
+  int nb[] = {1,     2,     4,     7,    11,    16,    23};
```

Comment:
This formatting is a little odd, but maybe Axel is going to run this through clang format?

Reply:
yes, among other things